### PR TITLE
DIV-4962 - Solicitor Personal Service

### PR DESCRIPTION
**Pull request by hosainnet**
# Description

Added the document for solicitor personal service, which is a letter generated when the a solicitor chooses that the respondent should be personally visited and notified of the divorce

Fixes #DIV-4962

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -50,7 +50,7 @@ withPipeline(type , product, component) {
 
     // Kubernetes does not retrieve variables from the output terraform
     before('functionalTest:preview') {
-        env.FEATURE_RESP_SOLICITOR_DETAILS = 'true'
+        env.FEATURE_RESP_SOLICITOR_DETAILS = 'false'
```
Comment (hosainnet) [inline Jenkinsfile_CNP]: Unreleased feature

File: src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/management/test/endpoint/DocumentGenerationTestController.java [hunk 2]
```diff
@@ -9,44 +9,41 @@
...
-                                                       + "pair of placeholder texts", required = true)
-                                                       Map<String, Object> allRequestParams) {
-        String templateName = (String) allRequestParams.remove(TEMPLATE_NAME);
-
-        byte[] generatedPDF = documentManagementService.generateDocument(templateName, allRequestParams);
+    public ResponseEntity generateDocument(
+                        @ApiParam(value = "JSON object containing the templateName and case details", required = true)
+                        @RequestBody
+                        @Valid
+                        GenerateDocumentRequest templateData) {
```
Comment (hosainnet) [inline src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/management/test/endpoint/DocumentGenerationTestController.java]: Take a proper request object just like the real endpoint

File: src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/management/test/endpoint/DocumentGenerationTestController.java [hunk 3]
```diff
@@ -9,44 +9,41 @@
...
+        byte[] pdf = documentManagementService.generateDocument(templateData.getTemplate(), templateData.getValues());
 
         return ResponseEntity
                 .ok()
-                .contentLength(generatedPDF.length)
-                .contentType(MediaType.APPLICATION_PDF)
-                .body(generatedPDF);
+                .contentLength(pdf.length)
+                .contentType(MediaType.APPLICATION_OCTET_STREAM)
```
Comment (hosainnet) [inline src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/management/test/endpoint/DocumentGenerationTestController.java]: This allows us to download the file from Swagger

File: src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/service/impl/EvidenceManagementServiceImpl.java [hunk 4]
```diff
@@ -50,6 +51,8 @@ public FileUploadResponse storeDocumentAndGetInfo(byte[] document, String author
             } else {
                 throw new DocumentStorageException("Failed to store document");
             }
+        } catch (HttpServerErrorException e) {
+            throw new DocumentStorageException("Error storing document " + e.getResponseBodyAsString(), e);
```
Comment (hosainnet) [inline src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/service/impl/EvidenceManagementServiceImpl.java]: Add better logged in the case of an error response

File: Jenkinsfile_CNP [hunk 5]
```diff
@@ -62,7 +62,5 @@ withPipeline(type , product, component) {
         steps.archiveArtifacts allowEmptyArchive: true, artifacts: '**/site/serenity/**/*'
     }
 
-    onMaster {
-        enableSlackNotifications(channel)
-    }
+    enableSlackNotifications(channel)
```
Comment (hosainnet) [inline Jenkinsfile_CNP]: To allow us to receive PR notification success/failures

File: src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/controller/DocumentGeneratorController.java [hunk 6]
```diff
@@ -51,4 +55,24 @@ public GeneratedDocumentInfo generatePDF(
...
+        @Valid
+            GenerateDocumentRequest templateData) {
+        byte[] pdf = documentManagementService.generateDocument(templateData.getTemplate(), templateData.getValues());
+
+        return ResponseEntity
+            .ok()
+            .contentLength(pdf.length)
+            .contentType(MediaType.APPLICATION_OCTET_STREAM)
+            .body(pdf);
+    }
```
Comment (hosainnet) [inline src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/controller/DocumentGeneratorController.java]: The above endpoint was a "test" one - it's now moved to the main controller as we need the ability to get generate the file but not upload it anywhere (e.g to attach in an email, for the purpose of this story)

File: Jenkinsfile_CNP (hunk 1 above)
Comment (alexwakeman) [reply Jenkinsfile_CNP]: this means test won't pass on AKS - as the feature is "on" on AKS...

File: charts/div-dgs/values.aat.template.yaml [hunk 7]
```diff
@@ -9,7 +9,7 @@ java:
         REFORM_ENVIRONMENT : "aat"
         DOCMOSIS_SERVICE_BASE_URL: "https://docmosis-development.platform.hmcts.net"
         MANAGEMENT_ENDPOINT_HEALTH_CACHE_TIMETOLIVE: "30000"
-        FEATURE_RESP_SOLICITOR_DETAILS: "true"
+        FEATURE_RESP_SOLICITOR_DETAILS: "false"
```
Comment (alexwakeman) [inline charts/div-dgs/values.aat.template.yaml]: Oh my bad you changed it here lol

File: src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/controller/DocumentGeneratorController.java [hunk 8]
```diff
@@ -37,8 +42,8 @@
             @ApiResponse(code = 500, message = "Returned when there is an unknown server error",
                     response = String.class)
         })
-    @PostMapping("/version/1/generatePDF")
-    public GeneratedDocumentInfo generatePDF(
+    @PostMapping("/generatePDF")
```
Comment (alexwakeman) [inline src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/controller/DocumentGeneratorController.java]: 👍

**Comment by nsingamsetty** [comment]
This is tested as part of DIv-4962. Merging now.

**Pull request merged**